src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -88,143 +88,184 @@ public InventorySync(@Nonnull final InventorySyncOptions syncOptions) {
      */
-    private CompletionStage<Void> processBatch(final List<InventoryEntryDraft> batchOfDrafts) {
+    private CompletionStage<Void> processBatch(@Nonnull final List<InventoryEntryDraft> batchOfDrafts) {
         return fetchExistingInventories(batchOfDrafts)
-                .thenCompose(existingInventories -> syncBatch(batchOfDrafts))
-                .exceptionally(exception -> {
-                    final String errorMessage = format(CTP_INVENTORY_FETCH_FAILED, extractSkus(batchOfDrafts));
-                    handleError(errorMessage, exception, batchOfDrafts.size());
-                    return null;
-                });
+            .thenCompose(oldInventoriesOptional -> oldInventoriesOptional
+                .map(oldInventories -> syncBatch(oldInventories, batchOfDrafts))
+                .orElseGet(() -> completedFuture(null)));
     }
 
     /**
      * Given a list of inventory entry drafts, this method extracts a list of all distinct SKUs from the drafts, then
-     * it tries to fetch a list of Inventory entries with those SKUs, then it populates the map of
-     * {@code existingInventories} with a {@link InventoryEntryIdentifier} as a key for each {@link InventoryEntry}
-     * of instances existing in a CTP project.
+     * it tries to fetch a list of Inventory entries with those SKUs. If operation succeed then {@link CompletionStage}
+     * containing {@link Optional} with fetched {@link List} is returned, otherwise an error is handled and
+     * {@link CompletionStage} with empty {@link Optional} is returned.
      *
-     * @param drafts {@link List} of inventory entry drafts to create mapping for.
-     * @return a future which contains an empty result after populating the map of {@code existingInventories} with
-     *         the instances of the existing {@link InventoryEntry} in the CTP project or an exception.
+     * @param drafts {@link List} of inventory entry drafts
+     * @return a future which contains an {@link Optional} that may contain list of inventory entries
      */
-    private CompletionStage<Void> fetchExistingInventories(@Nonnull final List<InventoryEntryDraft> drafts) {
+    private CompletionStage<Optional<List<InventoryEntry>>> fetchExistingInventories(@Nonnull final
+                                                                                     List<InventoryEntryDraft> drafts) {
         final Set<String> skus = extractSkus(drafts);
         return inventoryService.fetchInventoryEntriesBySkus(skus)
-                               .thenAccept(
-                                   existingEntries -> {
-                                       existingInventories = existingEntries
-                                           .stream().collect(toMap(InventoryEntryIdentifier::of,
-                                               entry -> entry));
-                                   });
+            .thenApply(Optional::of)
```

Comment:
I already answered the reason here https://github.com/commercetools/commercetools-sync-java/pull/58#discussion_r120898605